src/routes.js

```diff
@@ -7,17 +7,29 @@ import {
 } from "react-router-dom";
 import Registration from "./registration";
 import TestDescription from "./testDescription";
+import TestStart from "./testStart";
+import Result from "./result";
 import UserContext from "./userContext";
 
 export default class RouteAndSwitch extends Component {
-  setUserState = ({ firstName, lastName, age }) => {
-    this.setState({ firstName: firstName, lastName: lastName, age: age });
+  setUserState = ({ firstName, lastName, age, testDifficulty, testTime, questions }) => {
```

Comment:
Cred ca ai putea folosi direct context pentru a stoca datele necesare, nu? Pare ca le stochezi si in state, iar ulterior in context le trimiti doar cand faci acel submit.

Reply:
Am construit un state aici, pt a putea folosi `<UserContext.Provider value={this.state}>`, pt ca daca nu am value imi da eroare la Consumer. Sau am inteles gresit la ce te referi?